File: scripts/cyclic.py
#!/usr/bin/env python
"""The classic cyclic benchmark."""

import sys

from deepgroebner.ideals import cyclic
from deepgroebner.buchberger import buchberger


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: cyclic.py <number>")
    else:
        I = cyclic(int(sys.argv[1]))
        G = buchberger(I)
        print(len(G))

File: scripts/cyclic.cpp
#include <iostream>

#include "buchberger.h"
#include "ideals.h"


int main(int argc, char* argv[]) {
  if (argc != 2) {
    std::cout << "Usage: cyclic <number>" << std::endl;
    return 1;
  }
  auto F = cyclic(std::stoi(argv[1]));
  auto result = buchberger(F);
  std::cout << result.first.size() << std::endl;
}
